File: src/main/java/model/Americano.java
package model;

public class Americano extends Amortizacao {

    @Override
    protected void atualizarValores(Emprestimo loan) {
        juros = pagamento = saldo * loan.getTaxa();
    }

    @Override
    protected void ultimaParcela(Emprestimo loan) {
        amortizacao = loan.getMontante();
        pagamento = amortizacao + juros;
    }
}

File: docs/sistemas/Americano.js
import SistemaAmortizacao from "./SistemaAmortizacao.js";

export default class Americano extends SistemaAmortizacao {
    constructor() {
        super("Sistema Americano");
    }
    atualizarValores(montante, tempo, taxa) {
        this.juros = this.pagamento = this.saldo * taxa;
    }
    ultimaParcela(montante, tempo, taxa) {
        this.amortizacao = montante;
        this.pagamento = this.amortizacao + this.juros;
    }
}
